contracts/token/ERC677.cairo

```diff
@@ -0,0 +1,373 @@
+%lang starknet
```

Comment:
Unfortunately, Cairo doesn't support extensibility/inheritance yet. Therefore, this entire contract is essentially a copy of ERC20, with some added bits.